samples/test/externalclient.test.js

```diff
@@ -247,4 +349,41 @@ describe('samples for external-account', () => {
     // Confirm expected script output.
     assert.match(output, /DNS Info:/);
   });
+
+  it('should acquire ADC for AWS creds', async () => {
+    // Create AWS configuration JSON file.
+    const config = {
+      type: 'external_account',
+      audience: AUDIENCE_AWS,
+      subject_token_type: 'urn:ietf:params:aws:token-type:aws4_request',
+      token_url: 'https://sts.googleapis.com/v1beta/token',
+      service_account_impersonation_url:
+        'https://iamcredentials.googleapis.com/v1/projects/' +
+        `-/serviceAccounts/${clientEmail}:generateAccessToken`,
+      credential_source: {
+        environment_id: 'aws1',
+        regional_cred_verification_url:
+          'https://sts.{region}.amazonaws.com?Action=GetCallerIdentity&Version=2011-06-15',
+      },
+    };
+    await writeFile(configFilePath, JSON.stringify(config));
+
+    // Run sample script with GOOGLE_APPLICATION_CREDENTIALS environment
+    // variable pointing to the temporarily created configuration file.
+    // Populate AWS environment variables to simulate an AWS VM.
+    const output = await execAsync('node adc', {
```

Comment:
nit, I would be tempted to change this to `${process.execPath} adc`, just in case a bin is used other than Node.

Reply:
Thanks for the tip. This works great! It now also works on my machine too. I also injected the `process.env` into the `env` object. I had to pass it first so it doesn't overwrite the new value of `GOOGLE_APPLICATION_CREDENTIALS`.